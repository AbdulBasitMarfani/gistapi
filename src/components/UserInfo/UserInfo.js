import styled from "styled-components";
import IconContainer from "../../sharedComponets/IconContainer/IconContainer";

export const UserInfo = ({
  userName,
  userImage,
  files,
  forksUrl,
  comments,
  commentsUrl,
}) => {
  console.log("userImage: ", userImage);
  console.log("userName: ", userName);
  //   return userName;
  const fileCount = Object.keys(files).length;
  return (
    <Wrapper>
      <UserBaiscData>
        <ImageWrapper src={userImage} />
        <NameWrapper>{userName}</NameWrapper>
      </UserBaiscData>
      <IconWrapper>
        <IconContainer iconName="code" text={`${fileCount} Files`} />
        <IconContainer iconName="repo-forked" text="Forks" link={forksUrl} />
        <IconContainer
          iconName="comment"
          text={`${comments} Comments`}
          link={commentsUrl}
        />
        <IconContainer iconName="star" text="Stars" />
      </IconWrapper>
    </Wrapper>
  );
};
const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

const UserBaiscData = styled.div`
  display: flex;
  gap: 0.5rem;
  align-items: center;
`;

const ImageWrapper = styled.img.attrs((props) => ({
  src: props.src,
}))`
  width: 30px;
  height: 30px;
  border-radius: 50%;
`;

const NameWrapper = styled.span`
  display: block;
  font-size: 12px;
  color: var(--primary-color);
`;

const IconWrapper = styled.div`
  display: inline-flex;
  gap: 1rem;
`;
