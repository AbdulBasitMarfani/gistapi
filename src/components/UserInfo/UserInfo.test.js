import { render } from "@testing-library/react";
import { UserInfo } from "./UserInfo";
const files = {
  "readme.txt": {
    filename: "gistfile1.txt",
    raw_url:
      "https://gist.githubusercontent.com/dwfault/80059b65e85973dbd629dff8d041588d/raw/3599c7641b180f2a4b6041d5e0f2e38f073f0fe6/gistfile1.txt",
  },
  "script.txt": {
    filename: "test.xml",
    raw_url:
      "https://gist.githubusercontent.com/Bezarius/cf8d6d227abb4450fa548c15148d5285/raw/b8aad7d46618739da077c0e0a84d26f805f2f929/test.xml",
  },
};

describe("UserInfo", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <UserInfo
        userImage="https://avatars.githubusercontent.com/u/127468175?v=4"
        userName="Abdul Basit"
        files={files}
        forksUrl="https://api.github.com/gists/3d62c5bac006167e2bf09b938e172bef/forks"
        comments={0}
        commentsUrl="https://api.github.com/gists/3d62c5bac006167e2bf09b938e172bef/comments"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
