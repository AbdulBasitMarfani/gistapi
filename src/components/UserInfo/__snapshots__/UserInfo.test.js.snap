// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-kAyceB dYQapR"
  >
    <div
      class="sc-imWYAI jaoXtE"
    >
      <img
        class="sc-jXbUNg hjfduJ"
        src="https://avatars.githubusercontent.com/u/127468175?v=4"
      />
      <span
        class="sc-dhKdcB hACwmB"
      >
        Abdul Basit
      </span>
    </div>
    <div
      class="sc-kpDqfm brArDb"
    >
      <div
        class="sc-aXZVg iEJzdC"
      >
        <span
          class="octicon octicon-code"
        />
        <span
          class="sc-eqUAAy cYzGHt"
        >
          2 Files
        </span>
      </div>
      <div
        class="sc-aXZVg iEJzdC"
      >
        <span
          class="octicon octicon-repo-forked"
        />
        <a
          class="sc-gEvEer bGYWPw"
          href="https://api.github.com/gists/3d62c5bac006167e2bf09b938e172bef/forks"
          target="_blank"
        >
          Forks
        </a>
      </div>
      <div
        class="sc-aXZVg iEJzdC"
      >
        <span
          class="octicon octicon-comment"
        />
        <a
          class="sc-gEvEer bGYWPw"
          href="https://api.github.com/gists/3d62c5bac006167e2bf09b938e172bef/comments"
          target="_blank"
        >
          0 Comments
        </a>
      </div>
      <div
        class="sc-aXZVg iEJzdC"
      >
        <span
          class="octicon octicon-star"
        />
        <span
          class="sc-eqUAAy cYzGHt"
        >
          Stars
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
